TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -327,36 +331,50 @@ class RouteOptionsViewController: UIViewController {
         routeResults.reloadData()
     }
     
-    private func getDelay(tripId: String, stopId: String) -> Future<Response<Int?>> {
-        return networking(Endpoint.getDelay(tripID: tripId, stopID: stopId)).decode()
+    private func getAllDelays(trips: [Trip]) -> Future<Response<[Delay]>> {
+        return networking(Endpoint.getAllDelays(trips: trips)).decode()
     }
-
-    @objc func updateLiveTracking(sender: Timer) {
-        // For each route in each route array inside of the 'routes' array,
-        // retrieve its delay. Use index of route to save delay for route to
-        // JSON file.
+    
+    @objc func updateAllRoutesLiveTracking(sender: Timer) {
+        var trips: [Trip] = []
+        // For each route in each route array inside of the 'routes' array, get its
+        // tripId and stopId to create trip array for request to get all delays.
         for routesArray in routes {
-            for (index, route) in routesArray.enumerated() {
-                if route.isRawWalkingRoute() { return }
-                guard let direction = route.getFirstDepartRawDirection(),
-                    let tripId = direction.tripIdentifiers?.first,
-                    let stopId = direction.stops.first?.id else {
-                        return
+            for route in routesArray {
+                if !route.isRawWalkingRoute() {
+                    guard let direction = route.getFirstDepartRawDirection(),
+                        let tripId = direction.tripIdentifiers?.first,
+                        let stopId = direction.stops.first?.id else {
+                            return
+                    }
+                    tripDictionary[tripId] = route
+                    let trip = Trip(stopID: stopId, tripID: tripId)
+                    trips.append(trip)
                 }
-                getDelay(tripId: tripId, stopId: stopId).observe(with: { result in
-                    let fileName = "RouteTableViewCell"
-                    DispatchQueue.main.async {
-                        switch result {
-                        case .value (let delayResponse):
-                            guard delayResponse.data != nil, let delay = delayResponse.data else {
-                                return
+            }
+        }
```

Comment:
why do we have to recreate this `trips` array everytime this timer is triggered rather than just doing it once?

Reply:
oh ya, this was also a concern from when I was writing this too. I think it was mainly cause I wanted to make sure that we always had the updated routes array to render our trips array from and I wasn't sure when routes were generated but I moved it out so this should be fixed now!